api/packages.go

```diff
@@ -50,10 +50,31 @@ type InstalledPackage struct {
 	AvailableUpdate string
 }
 
+func isDirectory(path string) (bool, error) {
+	fileInfo, err := os.Stat(path)
+	if err != nil {
+		return false, err
+	}
+	return fileInfo.IsDir(), err
+}
+
 // Validate checks a package declaration validity against a Humio
 // server.
 func (p *Packages) Validate(viewName string, absDiretoryPath string) (*ValidationResponse, error) {
-	zipFilePath, err := createTempZipFromFolder(absDiretoryPath)
+	var zipFilePath string
+	var err error
+
+	isDir, err := isDirectory(absDiretoryPath)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if isDir {
+		zipFilePath, err = createTempZipFromFolder(absDiretoryPath)
+	} else {
+		zipFilePath = absDiretoryPath
```

Comment:
You can test whether the file is definitely a zip by `zip.OpenReader(absDirectoryPath)` and checking for nil error.

Reply:
You are VERY welcome to improve it. I don't have the time atm. This is ok for me.